### ISSUE TITLE
Global configuration file

### ISSUE THREAD

**Issue by KronosTheLate** [domluna/JuliaFormatter.jl#719]
From the docs:
"When [format](https://domluna.github.io/JuliaFormatter.jl/stable/api/#JuliaFormatter.format-Tuple{Any,%20JuliaFormatter.AbstractStyle}) called, it will look for .JuliaFormatter.toml in the location of the file being formatted"

It would be very nice to allow a global .JuliaFormatter.toml file, as one often wants the same style across projects. I guess currently one has to place the file in ones home directory? It would be nice to have it inside .julia instad for tidyness.

**Comment by domluna** [comment]
this should work already if you put a .JuliaFormatter.toml file in ~/julia.dev

**Comment by KronosTheLate** [comment]
That is not at all clear for me from https://domluna.github.io/JuliaFormatter.jl/dev/config/.

Are you saying that I should place `.JuliaFormatter.toml` inside `.julia/dev`? Most of my code does not have this as a parent directory. Namely, I generally put my julia code inside `~/GeekyStuff/Julia`, which does not have `.julia` as a parent folder.

What about making so that if no `.JuliaFormatter.toml` file is found in the current or parent directories, `.julia/.JuliaFormatter.toml` will be used?

**Comment by domluna** [comment]
> This function will look for .JuliaFormatter.toml in the location of the file being formatted, and searching up the file tree until a config file is (or isn't) found. When found, the configurations in the file will overwrite the given options. See [Configuration File](https://domluna.github.io/JuliaFormatter.jl/stable/config/#Configuration-File) for more details.


from https://domluna.github.io/JuliaFormatter.jl/stable/api/#JuliaFormatter.format-Tuple{Any}

**Comment by KronosTheLate** [comment]
Right, but I also have school related work in ~/Uni/morepath, so my home folder (or further up) is the only common parent folder, and I do not want to clutter my home folder with files as specific as this.

**Comment by penelopeysm** [comment]
I'm not really willing to have a blessed global config file in `$HOME` or in `JULIA_DEPOT_PATH` -- I'm not that keen on having a complex hierarchy of formatter configs because it increases the surface area for bugs and also makes it harder for people to report bugs in a reproducible way. I also don't really get where a global config file would live if not the home folder (plenty of tools happily clutter the home folder with extra files and dirs, or at best they'll use `~/.config`). However, I think a reasonable solution would be to allow specifying a fallback config file via an environment variable.
